src/sql/workbench/contrib/notebook/test/electron-browser/notebookModel.test.ts

```diff
@@ -865,9 +886,47 @@ suite('notebook model', function (): void {
 		assert.equal(output.metadata['multi_connection_mode'], true, 'multi_connection_mode not saved correctly to notebook metadata');
 	});
 
-	async function loadModelAndStartClientSession(): Promise<NotebookModel> {
+	test('Should not language info kernel alias name even if kernel spec is SQL', async function () {
 		let mockContentManager = TypeMoq.Mock.ofType(NotebookEditorContentManager);
-		mockContentManager.setup(c => c.loadContent()).returns(() => Promise.resolve(expectedNotebookContent));
+		mockContentManager.setup(c => c.loadContent()).returns(() => Promise.resolve(expectedKernelAliasNotebookContentOneCell));
+		defaultModelOptions.contentManager = mockContentManager.object;
+
+		queryConnectionService.setup(c => c.getActiveConnections(TypeMoq.It.isAny())).returns(() => null);
+
+		// Given I have a session that fails to start
+		sessionReady.resolve();
+
+		let model = new NotebookModel(defaultModelOptions, undefined, logService, undefined, new NullAdsTelemetryService(), queryConnectionService.object, configurationService);
```

Comment:
can we do `let model = loadModelAndStartclientSession(expectedKernelAliasNotebookContentOneCell);` here?

Reply:
So this was one of the issues I ran into. I was trying to do this but couldn't due to multiple errors.

We need the model.loadContents() in order for the metadata to properly set to the language_info to kusto and not the defaultKernel.

But there is a further issue if I use the function you state and the loadContents() as I get `TypeError: Cannot read property 'createInstance' of undefined` on [loadContents method](https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/services/notebook/browser/models/notebookModel.ts#L412). I'll look into this further on Test Mondays, to add more tests and do more revamping incrementally.